### PR TITLE
[Feature]: Added support to load openapi source and create rpcs for e…

### PR DESCRIPTION
**Pull request by khannaabhi**
…ach operation in openapi spec

### CODE REVIEW

File: sc-poc/Makefile [hunk 1]
```diff
@@ -24,5 +24,8 @@ build:
 install:
 	cd cmd/spacectl && go install
 
+install-deps:
```
Comment (YourTechBud) [inline sc-poc/Makefile:27]: Add this command in `.PHONY`

File: sc-poc/pkg/apis/core/v1alpha1/source_openapi.go [hunk 2]
```diff
@@ -22,6 +24,16 @@ type OpenAPISource struct {
...
+
+	// OpenAPI describes the value/ref for openapi schema
+	OpenAPI OpenAPISpec `json:"openApi"`
+}
+
+type OpenAPISpec struct {
+	// Value stores the open api schema
+	Value runtime.RawExtension `json:"value,omitempty"`
+	// Ref stores the path from which openapi schema will be fetched if value is not provided
+	Ref *HTTPOptions `json:"ref,omitempty"`
```
Comment (YourTechBud) [inline sc-poc/pkg/apis/core/v1alpha1/source_openapi.go:36]: Add kubebuilder comments for optional

File: sc-poc/sources/openapi/helpers.go [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+	cleanseSchemaRef(reqSchema)
+	cleanseSchemaRef(respSchema)
+	return &rpc.RPC{
+		Name:           operationID,
+		OperationType:  operationType,
+		RequestSchema:  reqSchema,
+		ResponseSchema: respSchema,
+		HTTPOptions: &v1alpha1.HTTPOptions{
+			Method: method,
+		},
```
Comment (YourTechBud) [inline sc-poc/sources/openapi/helpers.go:65]: No need for this

File: sc-poc/sources/openapi/source.go [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+func init() {
+	source.RegisterSource(Source{}, openapisourcesResource)
+}
+
+// Source describes the compiled openapi
+type Source struct {
+	v1alpha1.OpenAPISource
+
+	// Internal stuff
+	wg     *sync.WaitGroup
```
Comment (YourTechBud) [inline sc-poc/sources/openapi/source.go:32]: I don't think you need this

File: sc-poc/sources/openapi/source.go [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+				// export value of ext to plugins variable
+				if err := mapstructure.Decode(ext, &plugins); err != nil {
+					s.logger.Error("Unable to decode x-sc-plugins", zap.Error(err))
+				}
+			}
+			rpcs = append(rpcs, s.createRPC(url, getSpec.OperationID, operationType, http.MethodGet, respBody, reqBody, getSpec.Parameters, plugins))
+		}
+
+		// Process POST requests
+		if postSpec := pathSpec.Post; postSpec != nil {
```
Comment (YourTechBud) [inline sc-poc/sources/openapi/source.go:126]: Can we make this code DRY?

**Review by YourTechBud — CHANGES_REQUESTED** [review]

File: sc-poc/sources/openapi/helpers.go (hunk 3 above)
Comment (khannaabhi) [reply sc-poc/sources/openapi/helpers.go:65]: We need this or else we all the rpc will be exposed as POST endpoint. Is that the expected behavior?

Comment (YourTechBud) [reply sc-poc/sources/openapi/helpers.go:65]: Yes

File: sc-poc/sources/openapi/helpers.go [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+		url := fmt.Sprintf("%s%s", s.Spec.Source.URL, urlPath)
+
+		// Create the request
+		req, err := http.NewRequestWithContext(ctx, method, url, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		// Set request body for POST method
+		if method != http.MethodGet {
```
Comment (YourTechBud) [inline sc-poc/sources/openapi/helpers.go:88]:
```suggestion
		if method != http.MethodGet || method != http.MethodDelete {
```

File: sc-poc/sources/openapi/helpers.go (hunk 3 above)
Comment (YourTechBud) [inline sc-poc/sources/openapi/helpers.go:65]:
```suggestion
	return &rpc.RPC{
		Name:           operationID,
		OperationType:  operationType,
		RequestSchema:  reqSchema,
		ResponseSchema: respSchema,
```

**Review by YourTechBud — APPROVED** [review]

**Pull request merged**